demo/site/server.ts

```diff
@@ -34,16 +34,21 @@ app.prepare().then(() => {
                     return;
                 }
             }
-            const isAssetsPath = parsedUrl.pathname?.startsWith("/assets/");
-            if (
-                isAssetsPath ||
-                parsedUrl.pathname == "/favicon.ico" ||
-                parsedUrl.pathname == "/apple-icon.png" ||
-                parsedUrl.pathname == "/icon.svg" ||
-                parsedUrl.pathname == "/robots.txt" ||
-                parsedUrl.pathname == "/sitemap.xml"
-            ) {
-                const maxAge = isAssetsPath ? "604800" : "900"; // 1 week cache for /assets/*, otherwise 15 minutes
+
+            let maxAge: string | undefined;
+
+            if (parsedUrl.pathname?.startsWith("/assets/")) {
+                // assets in public/assets/* are cached for 1 week. When updated without changing the filename, the cache is not invalidated.
+                // To force an immediate update of a file, ensure the filename is changed as well
+                maxAge = "604800"; // 1 week cache
+            } else if (parsedUrl.pathname == "/favicon.ico" || parsedUrl.pathname == "/apple-icon.png" || parsedUrl.pathname == "/icon.svg") {
+                // images/icons in the /app folder automatically have a hash added to the filename, so they can be cached for a long time because the hash changes when the file content changes
+                maxAge = "31536000"; // 1 year cache
```

Comment:
who does that? does nextjs automatically inject the <link when the file exists? (and includes the hash)

Reply:
https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#icon
Yes, i now found something to the docs about it. Works only for icon and apple-icon

I checked here, I changed a small thing in the icon and it also changed the appended string:
https://github.com/user-attachments/assets/dcd71946-1415-4a38-b8d0-a1fd72ac6f7c